server/commands/src/impls.rs

```diff
@@ -186,3 +188,25 @@ pub fn whisper(
 
     Ok(())
 }
+
+#[command(usage = "say <message>")]
+pub fn say(ctx: &mut CommandCtx, message: TextArgument) -> anyhow::Result<()> {
+    let sender_name = if let Some(sender_name) = ctx.world.try_get::<Name>(ctx.sender) {
```

Comment:
`sender_name` could be an `&str`; there's not need to clone it from what I see.

Reply:
To add to this, you can also use `Option::as_ref` combined with `Option::unwrap_or_default`.

Reply:
> 
> 
> `sender_name` could be an `&str`; there's not need to clone it from what I see.

If you could show me what you would do to do this I would appreciate it. I've tried a few things and, other than placing the message generation within the `if let` statement, the borrow checker doesn't seem to appreciate not cloning the `String`.

Reply:
Sure thing:
```rust
let sender_name = ctx.world.try_get::<Name>(ctx.sender).map(|name| &name.0).unwrap_or("Server");
```
Does this work?

Reply:
Hmm, I see what the problem is... unfortunately `legion` has no `Ref::map` to fix this. I suppose performing the allocation is fine, for now.

Reply:
Alternatively, here's a hack:

```rust
let name = world.try_get::<Name>(ctx.sender);

let sender_name = if let Some(name) = name {
    &name.0
} else {
    None
};
```

Reply:
I also came up with `let sender_name = ctx.world.try_get::<Name>(ctx.sender).as_deref().unwrap_or(&Name(String::from("Server"))).0.clone();` as a "more succinct" way of handling the situation (it still clones the string, however)

Reply:
```rust
let name = Some(Name("player".to_owned()));
let name: &str = name.as_ref().map(|name| &*name.0).unwrap_or("server");
```

Reply:
Alternatively you can do
```rust
let name = Some(Name("player".to_owned()));
let name: &str = name.as_ref().map_or("server", |Name(n)| n);
```

Reply:
In every attempt to use &str I've tried, `ctx.world` gets borrowed for the lifetime of the `&str` and therefore I cant borrow it to `ctx.game.handle`

@Defman's suggestion seems to work for using the default name, but I can't use it because of the borrowing issue. 

[Here ](https://hastebin.com/vexopolulo.coffeescript) is the current code that I am using, I have indicated the spots where the borrow errors are occurring. Right now the only way I can see to avoid this is the clone.

Reply:
With my approach above, you can add `drop(name)` before the call to `handle` and everything should work fine.